### ISSUE TITLE
Crash after using enums in export variables.

### ISSUE THREAD

**Issue by AntonSalazar** [godotengine/godot#38168]
**Godot version:**
v3.2.1 b0eca5828ccd3a5c07b77e82e567c45937de9056
**OS/device including version:**
GNU Linux x64

**Issue description:**
When you declare variables export(enum) in the same class, and the enum take from another, when fast switching between nodes (or slow, but the result will be longer wait), then the crash of the engine.
The archive is a video demonstration of how to reproduce the bug, if something is unclear.

**Steps to reproduce:**
1) Create new project.
2) Create 3 GDScripts: "ScriptA", "ScriptB, "ScriptC" and add "class_name" for each of them.
3) "ScriptA" - it contains a enum (E_TEST_ENUM {ZERO = 0, ONE = 1, TWO = 2, THREE = 3})
4) "ScriptB" - contains export(ScriptA.E_TEST_ENUM) var _test_enum: int
5) "ScriptC" - extends from "ScriptB". Can contain anything.
6) Create a simple scene: any root node and multiple nodes with a "ScriptC" GDscript.
7) Switch between nodes. Preferably quickly.

**Minimal reproduction project:**
[ExportEnumsBug.zip](https://github.com/godotengine/godot/files/4526607/ExportEnumsBug.zip)

**Comment by MWFIAE** [comment]
I couldn't reproduce it with v3.2.1.stable.mono on Windows

**Comment by Ev1lbl0w** [comment]
Was able to reproduce on 3.2.1-stable but **not** on branch master. Maybe it was already fixed..?

**Comment by AntonSalazar** [comment]
> Was able to reproduce on 3.2.1-stable but not on branch master. Maybe it was already fixed..?

What is the point of checking the master branch if they actively do 4.0?
I last checked on commit ed27b7e6b973eb6be6fdac4439e99a1907b9cb58, and this is 3.2.2 already. This bug is reproduced in the same way.

**Comment by akien-mga** [comment]
> What is the point of checking the main branch if they are actively doing 4.0? If I'm not mistaken.
I last checked on commit ed27b7e, and this is 3.2.2 already. This bug is reproduced in the same way.

Well the point is to give information that this bug has been fixed in the development branch, and will be fixed in 4.0 and later versions. The commit that fixed it could be identified and backported to the 3.2 branch, thus also fixing it for future 3.2.x versions.

**Comment by AntonSalazar** [comment]
I checked on the current last commit 510406af28b9a4f101db631bb256c7ede2966c3d in branch 3.2 - the result is the same.

**Comment by Ev1lbl0w** [comment]
I've retested, and after all it's also happening on the `master` branch. Looks like a problem on the GDScript parser, but I have no experience here to help.

Here's a debug stacktrace, might be helpful:
```
handle_crash: Program crashed with signal 11
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] /usr/lib/libc.so.6(+0x3bd70) [0x7efe372cdd70] (??:0)
[2] /usr/lib/libstdc++.so.6(__dynamic_cast+0x21) [0x7efe3765bd11] (??:0)
[3] GDScriptNativeClass* Object::cast_to<GDScriptNativeClass>(Object*) (/home/ricardo/Workspace/Programs/godot/./core/object.h:589)
[4] GDScriptParser::_parse_class(GDScriptParser::ClassNode*) (/home/ricardo/Workspace/Programs/godot/modules/gdscript/gdscript_parser.cpp:4542)
[5] GDScriptParser::_parse(String const&) (/home/ricardo/Workspace/Programs/godot/modules/gdscript/gdscript_parser.cpp:8532)
[6] GDScriptParser::parse(String const&, String const&, bool, String const&, bool, Set<int, Comparator<int>, DefaultAllocator>*, bool) (/home/ricardo/Workspace/Programs/godot/modules/gdscript/gdscript_parser.cpp:8637)
[7] GDScriptLanguage::get_global_class_name(String const&, String*, String*) const (/home/ricardo/Workspace/Programs/godot/modules/gdscript/gdscript.cpp:1954)
[8] EditorData::script_class_get_base(String const&) const (/home/ricardo/Workspace/Programs/godot/editor/editor_data.cpp:905)
[9] EditorData::script_class_get_icon_path(String const&) const (/home/ricardo/Workspace/Programs/godot/editor/editor_data.cpp:941)
[10] EditorNode::get_object_icon(Object const*, String const&) const (/home/ricardo/Workspace/Programs/godot/editor/editor_node.cpp:3753)
[11] EditorPath::update_path() (/home/ricardo/Workspace/Programs/godot/editor/editor_path.cpp:93)
[12] InspectorDock::update(Object*) (/home/ricardo/Workspace/Programs/godot/editor/inspector_dock.cpp:390)
[13] EditorNode::_edit_current() (/home/ricardo/Workspace/Programs/godot/editor/editor_node.cpp:1985)
[14] EditorNode::push_item(Object*, String const&, bool) (/home/ricardo/Workspace/Programs/godot/editor/editor_node.cpp:?)
[15] SceneTreeDock::_node_selected() (/home/ricardo/Workspace/Programs/godot/editor/scene_tree_dock.cpp:1213)
[16] void call_with_variant_args_helper<SceneTreeDock>(SceneTreeDock*, void (SceneTreeDock::*)(), Variant const**, Callable::CallError&, IndexSequence<>) (/home/ricardo/Workspace/Programs/godot/./core/callable_method_pointer.h:140)
[17] void call_with_variant_args<SceneTreeDock>(SceneTreeDock*, void (SceneTreeDock::*)(), Variant const**, int, Callable::CallError&) (/home/ricardo/Workspace/Programs/godot/./core/callable_method_pointer.h:162)
[18] CallableCustomMethodPointer<SceneTreeDock>::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/ricardo/Workspace/Programs/godot/./core/callable_method_pointer.h:178)
[19] Callable::call(Variant const**, int, Variant&, Callable::CallError&) const (/home/ricardo/Workspace/Programs/godot/core/callable.cpp:50)
[20] MessageQueue::_call_function(Callable const&, Variant const*, int, bool) (/home/ricardo/Workspace/Programs/godot/core/message_queue.cpp:253)
[21] MessageQueue::flush() (/home/ricardo/Workspace/Programs/godot/core/message_queue.cpp:305)
[22] SceneTree::iteration(float) (/home/ricardo/Workspace/Programs/godot/scene/main/scene_tree.cpp:423)
[23] Main::iteration() (/home/ricardo/Workspace/Programs/godot/main/main.cpp:2151)
[24] OS_LinuxBSD::run() (/home/ricardo/Workspace/Programs/godot/platform/linuxbsd/os_linuxbsd.cpp:259)
[25] ./bin/godot.linuxbsd.tools.64.llvm(main+0x1a8) [0x33e7068] (/home/ricardo/Workspace/Programs/godot/platform/linuxbsd/godot_linuxbsd.cpp:56)
[26] /usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7efe372b9023] (??:0)
[27] ./bin/godot.linuxbsd.tools.64.llvm(_start+0x2e) [0x33e6dfe] (??:?)
```

**Comment by akien-mga** [comment]
Fixed by #42936.

**Comment by AntonSalazar** [comment]
Many thanks!
